### DISCUSSION TITLE
How to use the Python API to generate the changelog?

### DISCUSSION THREAD

**Discussion by ayush9pandey** [pawamoy/git-changelog#45] [Q&A] [answered]
I am able to use this package on command line and it generates a changelog. However, when using the `build_and_render` function as described in the documentation, I don't get the full changelog. Instead, I only get the version name with the comparison link that is added to the CHANGELOG.md file. 

I would appreciate any support with this. 

Here's how I am running it:

```
out1, out2 = build_and_render(
    repository=".",
    output="CHANGELOG.md",
    convention="angular",
    template="keepachangelog",
    parse_trailers=True,
    parse_refs=False,
    sections=("build", "deps", "feat", "fix", "refactor"),
    bump_latest=True,
    in_place=True,
    marker_line="<!-- insertion marker -->",
    version_regex= r"^## \[v?(?P<version>[^\]]+)",
)
```

**Comment by pawamoy** [comment]
Try setting `in_place` to `False`: rendering in-place will only insert new contents, not render the whole changelog.
With `in_place=False`, you can also remove the `marker_line` and `version_regex` parameters, which are now unused.

**Comment by ayush9pandey** [comment]
Thanks for the reply! I tried that but unfortunately I think I am still not getting the expected output. Let me try to elaborate:

1. When I run `git-changelog` using the command line, I get a long changelog output in the CLI with actual changes to the package that is divided into four sections "Added" "Fixed" "Changed" "Removed". These sections are placed after the "Compare with latest" links. 
2. When I run the `build_and_render` function, I only get the "compare with latest" links added to the CHANGELOG.md file. 

I want the CLI behavior but using the API function. So, I'd like the sections of the changes added to the change log file.

**Comment by pawamoy** [comment, marked as answer]
Thanks for elaborating :slightly_smiling_face: 
Then it seems like you're using the wrong commit message convention: try using `convention="basic"` instead.

**Comment by ayush9pandey** [comment]
Thanks, that worked! I appreciate the help. Could you explain how the different conventions work (and why the angular did not work), or if there's a link that you can point to that I can read more about it.

**Reply by pawamoy** [reply]
Checkout our docs: https://pawamoy.github.io/git-changelog/usage/#choose-the-commit-message-convention :slightly_smiling_face:
